canopy_bioparm_mod.F90

```diff
@@ -0,0 +1,802 @@
+module canopy_bioparm_mod
+
+    implicit none
+
+contains
+
+!:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+    SUBROUTINE CANOPY_BIOP( EMI_IND, LU_OPT, VTYPE, &
+        EF, CT1, CEO)
+
+!-----------------------------------------------------------------------
+
+! Description:
+!     gets biogenic emissions factors and parameters from MEGAN2.1
+
+! Preconditions:
+!     emissions index, lu option, and vegtype
+
+! Subroutines and Functions Called:
+
+! Revision History:
+!     Prototype 02/23 by PCC, based on Guenther et al. (2012)
+! Citation:
+!Guenther, A. B., et al.,: The Model of Emissions of Gases and Aerosols from
+!Nature version 2.1 (MEGAN2.1): an extended and updated framework for
+!modeling biogenic emissions, Geosci. Model Dev., 5, 1471–1492,
+!https://doi.org/10.5194/gmd-5-1471-2012, 2012.
+!-----------------------------------------------------------------------
+!-----------------------------------------------------------------------
+!     Feb 2023 P.C. Campbell: Initial version
+!-----------------------------------------------------------------------
+!-----------------------------------------------------------------------
+
+        use canopy_const_mod, ONLY: rk
+
+! Arguments:
+!     IN/OUT
+        INTEGER,     INTENT( IN )       :: EMI_IND         ! Input biogenic emissions index
+        INTEGER,     INTENT( IN )       :: LU_OPT          ! integer for LU type from model mapped to Massman et al. (default = 0/VIIRS)
+        INTEGER,     INTENT( IN )       :: VTYPE           ! Grid cell dominant vegetation type
+        REAL(RK),    INTENT( OUT )      :: EF              ! Out Mapped EF ((ug/m2 hr)
+        REAL(RK),    INTENT( OUT )      :: CT1             ! Out Activation energy (kJ/mol)
+        REAL(RK),    INTENT( OUT )      :: CEO             ! Out Empirical coefficient
+!  LOCAL
+        REAL(RK) :: EF1,EF2,EF3,EF4,EF5,EF6,EF7    ! Plant Emission factors (EF) (ug/m2 hr)
+        REAL(RK) :: EF8,EF9,EF10,EF11,EF12,EF13    !
+        REAL(RK) :: EF14,EF15                      !
+
+! Plant-Dependent emissions capacity/factors (EFs) for Isoprene (Tables 2-3 of Guenther et al., 2012) (ug/m2 hr)
+        REAL(RK),          PARAMETER     :: EF1_ISOP    =  600.0_rk     ! Needleleaf Evergreen Temperate Tree
```

Comment:
could place the EFs in a single 2-d array with species dimension

Reply:
@zmoon   Yes, seems more compact and cleaner as your initial comment on the 4-d emissions array.  Can you go ahead and add these 2-d EF changes to the new branch with the 4-d emi_bio array as well?